cijoe/scripts/bench_plotter.py

```diff
@@ -80,86 +72,99 @@ def plot_attributes_from_step(step):
 
     limits_path = Path(step.get("with", {}).get("limits", "plot-limits.yaml"))
     legends_path = Path(step.get("with", {}).get("legends", "plot-legends.yaml"))
+    styles_path = Path(step.get("with", {}).get("styles", "plot-styles.yaml"))
 
     return {
         "limits": dict_from_yamlfile(limits_path),
         "legends": dict_from_yamlfile(legends_path),
+        "styles": dict_from_yamlfile(styles_path),
     }
 
 
-def lineplot(args, cijoe, step):
-    search = step.get("with", {}).get("path", args.output)
-    if not search:
-        return errno.EINVAL
+def draw_bar_plot(data, plot_attributes, xlabel=None, y_limit=None):
+    colors = plot_attributes["styles"].get("colors", ["#000000"])
+    hatches = plot_attributes["styles"].get("hatches", ["."])
+    width = 0.20
 
-    tool = step.get("with", {}).get("tool", "fio")
-    if tool == "fio":
-        search_for = FIO_OUTPUT_NORMALIZED_FILENAME
-    elif tool == "bdevperf":
-        search_for = BDEVPERF_OUTPUT_NORMALIZED_FILENAME
+    groups = list(set(map(lambda item: item["group"], data.values())))
+    is_grouped = len(groups) > 1
 
-    path = next(Path(search).rglob(search_for))
-    with path.open() as jfd:
-        data = json.load(jfd)
+    num_bars = len(data.items()) - 1
+    if is_grouped:
+        x_ticks = dict()
+        x_tick_info = dict([(group, [groups.index(group), 0]) for group in groups])
+    else:
+        unique_x_values = map(lambda xy: xy[0], list(data.values())[0]["xys"])
+        x_ticks = dict(
+            [(x, i + (num_bars * width) / 2) for i, x in enumerate(unique_x_values)]
+        )
 
-    plot_attributes = plot_attributes_from_step(step)
+    if y_limit is None:
+        # assumes that the highest x-value will also have the highest y value.
+        # for each group, take y-value of the last (x,y) coordinate, and take the
+        # max value of these.
+        y_limit = max(map(lambda item: item["xys"][-1][1], data.values())) * 1.3
```

Comment:
I think that is notation from the numpy library. A list of (x,y) coordinates can't be mapped to just its y-values with the `xys[:][1]` notation. This will return the (x,y) coordinate on index 1.
The assumption is made to have a bit cleaner code, but to get rid of it, I would have to write something like
`max(map(lambda item: max(map(lambda xy: xy[1], item["xys"])), data.values()))`

Reply:
Ah you're right. I think `max(map(lambda item: max(map(lambda xy: xy[1], item["xys"])), data.values()))` could be acceptable with a comment along the lines of "Get the max y-value across all plots"